AutoSteamApp/Automaton/SteamworkAutomaton.cs

```diff
@@ -0,0 +1,348 @@
+                    _SteamworksData = new SteamworksData(_Process);
+                    _SaveData = new SaveData(_Process);
+                }
+                _InputSimulator = new InputSimulator();
+            }
+            catch (Exception e)
+            {
+                Log.Exception(new Exception("Failed to initialze Steamworks Automaton\n\t", e));
+                Log.Warning("Something went wrong reading the MHW:IB process.");
+                if (!ConfigurationReader.ShouldAutoQuit)
+                {
+                    Log.Message("Press any key to exit.");
+                    Console.ReadKey();
+                }
+                Environment.Exit(-1);
+
+            }
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Performs the main logic loop of reading the steamworks sequence, 
+        /// then performing actions depending on the configurations.
+        /// </summary>
+        /// <param name="cts">Cancellation token used to signal an exit.</param>
+        public void Run(CancellationToken cts)
+        {
+            try
+            {
+                while (!cts.IsCancellationRequested)
+                {
+                    if (_SupportedVersion && !ConfigurationReader.RandomRun)
+                    {
+                        // If we have satisfied all exit conditions
+                        if (CheckForExitCondition())
+                            // If we should auto exit, quit immediately
+                            if (ConfigurationReader.ShouldAutoQuit)
+                                Environment.Exit(0);
+                            // Otherwise wait for quit to be typed
+                            else
+                                return;
+
+                        // Otherwise we need to extract the sequence
+                        ExtractAndEnterSequence(cts);
+
+                        // Then check the steamworks state
+                        CheckSteamworksState(cts);
+                    }
+                    else
+                    {
+                        EnterRandomSequence(cts);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Exception(new Exception("Failed automating steamworks\n\t", e));
+                Log.Warning("Something went wrong trying to automate the steamworks.");
+                // If we should auto exit, quit immediately
+                if (ConfigurationReader.ShouldAutoQuit)
+                    Environment.Exit(-1);
+                // Otherwise wait for quit to be typed
+                else
+                    return;
+            }
+        }
+
+        private void EnterRandomSequence(CancellationToken cts)
+        {
+            // Generate a sequence to input
+            VirtualKeyCode[] sequence;
+            // If the version is unsuported, use a random sequence
+            sequence = StaticHelpers.RandomSequence();
```

Comment:
It is a good practice to name methods by what they do - here you could use GetRandomSequence() - also, why not initialize it directly when declaring the variable ?